module/apmhttp/handler.go

```diff
@@ -42,7 +42,7 @@ func Wrap(h http.Handler, o ...ServerOption) http.Handler {
 		handler:        h,
 		tracer:         apm.DefaultTracer,
 		requestName:    ServerRequestName,
-		requestIgnorer: DefaultServerRequestIgnorer(),
+		requestIgnorer: DynamicServerRequestIgnorer(apm.DefaultTracer),
```

Comment:
We'll need to set the default requestIgnorer after running the option functions (and only if a requestIgnorer is not supplied through WithRequestIgnorer), in case a non-default tracer is supplied with WithTracer.